kind/pkg/cluster/config/types.go

```diff
@@ -0,0 +1,50 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+// NOTE: if you change these types you likely need to update
+// Validate() and DeepCopy() at minimum
+
+// Config contains cluster creation config
+// This is the current internal config type used by cluster
+// Other API versions can be converted to this struct with Convert()
+type Config struct {
+	// NumNodes is the number of nodes to create (currently only one is supported)
+	NumNodes int `json:"numNodes,omitempty"`
+	// KubeadmConfigTemplate allows overriding the default template in
+	// cluster/kubeadm
+	KubeadmConfigTemplate string `json:"kubeadmConfigTemplate"`
```

Comment:
yeah debating that, I was thinking we should mark this `omitempty` but then we'll never actually have it empty because we'll apply defaults at loading and empty will yield the default template. I guess just in case someone constructs a raw struct and does not call apply defaults.

Reply:
added omitempty